src/System.Private.ServiceModel/tests/Common/Scenarios/Endpoints.cs

```diff
@@ -1,49 +1,183 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.ServiceModel;
+using System.Security.Cryptography.X509Certificates;
+
 using Infrastructure.Common;
 
 public static partial class Endpoints
 {
+    private static X509Certificate2 rootCert;
+    private static X509Certificate2 clientCert;
+    private static object rootCertLock = new object();
+    private static object clientCertLock = new object();
+    //Install Root certificate, this is required for all https test
+    private static void EnsureRootCertificateInstalled()
+    {
+        lock (rootCertLock)
+        {
+            if (rootCert == null)
+            {
```

Comment:
Not blocking:  For future consideration we might let the first test hit and report the failure and then set a value so the remaining [ConditionalFact] tests depending on the root cert show as Skipped due to no root cert. This will satisfy @iamjasonp 's suggestion not to attempt doing it again if it fails once.